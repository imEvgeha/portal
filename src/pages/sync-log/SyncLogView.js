import React from 'react';
import PropTypes from 'prop-types';
import ArrowLeftIcon from '@atlaskit/icon/glyph/arrow-left';
import './SyncLogView.scss';
import {NexusTitle} from '../../ui/elements';
import SyncLogTable from './components/sync-log-table/SyncLogTable';
import {TITLE} from './syncLogConstants';

const SyncLogView = ({history}) => {
    return (
        <div className="nexus-c-sync-log-view">
            <NexusTitle>
                <button className="nexus-c-sync-log-view__back-btn" onClick={() => history.goBack()}>
                    <ArrowLeftIcon size="large" />
                </button>
                <span className="nexus-c-sync-log-view__title">{TITLE}</span>
            </NexusTitle>
            <div className="nexus-c-sync-log-view__table">
                <SyncLogTable />
            </div>
        </div>
    );
};

SyncLogView.propTypes = {
    history: PropTypes.object,
};

SyncLogView.defaultProps = {
    history: {},
};

export default SyncLogView;
