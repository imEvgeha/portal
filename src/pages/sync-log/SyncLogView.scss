.nexus-c-sync-log-view {
  display: flex;
  flex-direction: column;
  height: 100%;
  padding: 24px 0 0;
}

.nexus-c-sync-log-view__table {
    width: 100%;
    height: 100%;
    padding: 24px 24px 45px 24px;

    & .nexus-c-nexus-grid {
        height: 100%;
    }
}
