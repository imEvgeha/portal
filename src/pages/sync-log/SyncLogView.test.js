import React from 'react';
import {shallow} from 'enzyme';
import SyncLogView from './SyncLogView';

describe('SyncLogView', () => {
    describe('HTML content', () => {
        let wrapper = null;
        let syncLogTableWrapper = null;

        beforeEach(() => {
            wrapper = shallow(<SyncLogView />);
            syncLogTableWrapper = wrapper.find('SyncLogTable');
        });

        it('should render Sync Log view title', () => {
            expect(wrapper.find('.nexus-c-sync-log-view__title').text()).toEqual('Sync Log');
        });

        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('should render SyncLogTable', () => {
            expect(syncLogTableWrapper.length).toEqual(1);
        });
    });
});
