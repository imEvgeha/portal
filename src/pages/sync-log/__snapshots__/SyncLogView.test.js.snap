// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncLogView HTML content should match snapshot 1`] = `
<div
  className="nexus-c-sync-log-view"
>
  <NexusTitle
    className=""
    isInline={false}
    isSubTitle={false}
    title={null}
  >
    <Link
      replace={false}
      to="/metadata"
    >
      <ArrowLeftIcon
        size="large"
      />
    </Link>
    <span
      className="nexus-c-sync-log-view__title"
    >
      Sync Log
    </span>
  </NexusTitle>
  <div
    className="nexus-c-sync-log-view__table"
  >
    <Connect(SyncLogTable) />
  </div>
</div>
`;
